src/utils/http-transport.ts

```diff
@@ -0,0 +1,102 @@
+type DataProps = XMLHttpRequestBodyInit;
+
+type MethodProps = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+type OptionsProps = {
+  method: MethodProps;
+  timeout?: number;
+  data?: DataProps;
+  headers?: { [key: string]: string };
+};
+
+const METHODS: { [k: string]: MethodProps } = {
+  GET: 'GET',
+  POST: 'POST',
+  PUT: 'PUT',
+  DELETE: 'DELETE',
+};
```

Comment:
Для описания списков констант в TS существует такой инструмент, как [enum](https://www.typescriptlang.org/docs/handbook/enums.html):
```suggestion
enum HttpMethods {
    GET = 'GET',
    POST = 'POST',
    PUT = 'PUT',
    DELETE = 'DELETE',
};
```
В качестве типа для `MethodProps` можно использовать `HttpMethods`